### ISSUE TITLE
Seeding your pseudo-random state

### ISSUE THREAD

**Issue by ricardodeazambuja** [mila-iqia/ivado-mila-dl-school-2019#2]
I know I'm probably being picky here, but since it's a tutorial I suppose it's a good idea to fix or explain it better. In the notebook Data_tutorial_to_complete.ipynb, there's this:
```np.random.seed(seed)```
However, without enforcing the use of something like the [RandomState object](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.random.RandomState.html), there's no guarantee that another piece of code will not seed it again. Long time ago, I wasted some good hours (or days?) debugging a piece of code that should be deterministic, but it was not exactly because of the use of np.random.seed 😬

**Comment by gmarceaucaron** [comment]
Thank you for the comment. Indeed, using a local `RandomState` object gives you more control, and so you can avoid side effects like the `RandomState` being reinitialized by third-party code. However, in practice, we need to initialize three RNGs (Numpy, Pytorch CPU, and Pytorch GPU). Ensuring that your three RNGs are used everywhere in Pytorch can be difficult. Indeed, contrary to Scikit-Learn, essential functions with randomness (e.g., initialization) do not take a `RandomState` object as an argument. In the end, you can have some modules using your local RNGs and others using the global RNGs. An easy way to avoid it is to use only global RNGs. In the Pytorch documentation on Reproducibility, they only show how to [use global RNGs](https://pytorch.org/docs/stable/notes/randomness.html).
